src/main/scala/cromwell/binding/package.scala

```diff
@@ -24,17 +31,38 @@ package object binding {
   type CallInputs = Map[String, WdlValue]
   type CallOutputs = Map[LocallyQualifiedName, WdlValue]
   type HostInputs = Map[String, WdlValue]
-
   type ImportResolver = String => WdlSource
 
   /**
-   * Core data identifying a workflow including its unique ID, its namespace, and strongly typed inputs.
+   * Constructs a representation of a particular workflow invocation
    */
-  case class WorkflowDescriptor(id: WorkflowId, namespace: NamespaceWithWorkflow, wdlSource: WdlSource, wdlJson: WdlJson, actualInputs: WorkflowCoercedInputs) {
+  case class WorkflowDescriptor(id: WorkflowId, sourceFiles: WorkflowSourceFiles) {
+    val workflowOptions = Try(sourceFiles.workflowOptionsJson.parseJson) match {
+      case Success(JsObject(options)) =>
+        if (options.values.nonEmpty && !options.values.exists(_.isInstanceOf[JsString])) {
+          throw new Throwable(s"Workflow ${id.toString} options JSON is not a String -> String map: ${sourceFiles.workflowOptionsJson}")
+        }
+        options map { case (k, v) => k -> v.asInstanceOf[JsString].value }
+      case _ => throw new Throwable(s"Workflow ${id.toString} contains bad workflow options JSON: ${sourceFiles.inputsJson}")
+    }
+
+    val backendType = Backend.from(workflowOptions.getOrElse("default_backend", ConfigFactory.load.getConfig("backend").getString("backend")))
+    val namespace = NamespaceWithWorkflow.load(sourceFiles.wdlSource, backendType.backendType)
     val name = namespace.workflow.name
     val shortId = id.toString.split("-")(0)
+
+    val rawInputs = Try(sourceFiles.inputsJson.parseJson) match {
+      case Success(JsObject(inputs)) => inputs
+      case _ => throw new Throwable(s"Workflow ${id.toString} contains bad inputs JSON: ${sourceFiles.inputsJson}")
+    }
+
+    val coercedInputs = namespace.coerceRawInputs(rawInputs).get
```

Comment:
I'd argue that if the contract is "construction of objects can throw exceptions" then I think the use of `.get` here is just fulfilling a contract and not acting as an unsafe usage of `.get`

I'd call an unsafe `.get` is one in which:

* Doesn't properly consider all possible results of `.get`
* Makes a faulty assumption about some underlying component.

I don't think this code violating either of those.  If I were to get rid of the `.gets` but keep the same contract I'd just end up with superfluous code like:

```scala
namespace.staticDeclarationsRecursive(coercedInputs).getOrElse(throw new ...)
```

We can certainly debate whether or not exceptions should be thrown in this context at all or if we should restructure it into something that always returns a `Try[WorkflowDescriptor]`.  Personally I don't really care all that much (which is why I just chose to follow suit with other constructors).